lambdas/services/virus_scan_result_service.py

```diff
@@ -0,0 +1,133 @@
+        self.ssm_service = SSMService()
+        self.username = ""
+        self.password = ""
+        self.base_url = ""
+        self.access_token = ""
+
+    def prepare_request(self, file_ref):
+        try:
+            self.get_ssm_parameters_for_request_access_token()
+            self.virus_scan_request(file_ref, retry_on_expired=True)
+        except ClientError as e:
+            logger.error(
+                f"{LambdaError.VirusScanAWSFailure.to_str()}: {str(e)}",
+                {"Result": FAIL_SCAN},
+            )
+            raise VirusScanResultException(500, LambdaError.VirusScanAWSFailure)
+
+    def virus_scan_request(self, file_ref: str, retry_on_expired: bool):
+        try:
+            headers = {
+                "Content-Type": "application/json",
+                "Authorization": "Bearer " + self.access_token,
+            }
+            scan_url = self.base_url + "/api/Scan/Existing"
+            json_data_request = {
+                "container": self.staging_s3_bucket_name,
+                "objectPath": file_ref,
+            }
+            logger.info(json_data_request)
+
+            response = requests.post(
+                url=scan_url, data=json.dumps(json_data_request), headers=headers
+            )
+            if response.status_code == 401 and retry_on_expired:
+                self.get_new_access_token()
+                return self.virus_scan_request(file_ref, retry_on_expired=False)
+            response.raise_for_status()
+
+            parsed = response.json()
+
+            if parsed["result"] == "Clean":
+                logger.info(
+                    "Virus scan request succeeded",
+                    {"Result": "Virus scan request succeeded"},
+                )
+                return
+            else:
+                logger.info(
+                    "File is not clean",
+                    {"Result": FAIL_SCAN},
+                )
+                raise VirusScanResultException(400, LambdaError.VirusScanUnclean)
+
+        except HTTPError:
+            logger.info(
+                "Virus scan request failed",
+                {"Result": FAIL_SCAN},
+            )
+            raise VirusScanResultException(400, LambdaError.VirusScanTokenRequest)
+
+    def get_new_access_token(self):
+        try:
+            logger.info(f"username: {self.username}")
+            logger.info(f"password: {self.password}")
+
+            json_login = json.dumps(
+                {"username": self.username, "password": self.password}
+            )
+            token_url = self.base_url + "/api/Token"
+
+            response = requests.post(
+                url=token_url,
+                headers={"Content-type": "application/json"},
+                data=json_login,
+            )
+
+            response.raise_for_status()
+            logger.info(response.json())
+            new_access_token = response.json()["accessToken"]
```

Comment:
Do we need any protections in the event accessToken is not returned?

Reply:
added a key error to the except